commons/all/src/main/java/org/infinispan/commons/dataconversion/MediaType.java

```diff
@@ -318,6 +319,15 @@ public String toStringExcludingParam(String... params) {
       return builder.append("; ").append(strParams).toString();
    }
 
+   /**
+    * @return true if the MediaType's java type is a byte array.
+    */
+   public boolean isBinary() {
```

Comment:
So to clarify we assume everything is binary if it has ByteArray as a type in the media type (ie. application/object;type=ByteArray) or it doesn't match application object?

Reply:
Media types like text/* wouldn't be binary, although they could support size based eviction.

Reply:
Also I am not sure for application/unknown etc. 

To be honest thinking more we may need pluggable size calculators based on media type moving forward.

Reply:
The idea is that everything is binary by default. ```text/plain``` is UTF-8 ```byte[]```, same as ```application/json```. The rules are coded in ```MediaType.isBinary```.
Regarding size calculators, it's enough to know the java type from the MediaType, right?

Reply:
> The idea is that everything is binary by default.

Okay that was the part I was unaware.

I guess in that case the size calculators are probably fine then how they are.